--- liboil-0.3.1/configure.ac.orig	2005-03-17 19:44:51.000000000 +0100
+++ liboil-0.3.1/configure.ac	2005-03-21 20:53:43.000000000 +0100
@@ -147,6 +147,7 @@
 AC_CHECK_HEADER(ieee754.h,
   AC_DEFINE(HAVE_IEEE754_H, 1, [Define if ieee754.h exists]))
 
+LIBOIL_CFLAGS="$CFLAGS"
 AS_COMPILER_FLAG(-Wall, LIBOIL_CFLAGS="$LIBOIL_CFLAGS -Wall")
 if test "x$LIBOIL_CVS" = "xyes"
 then
@@ -163,7 +164,7 @@
     true)
 fi
 
-LIBOIL_CFLAGS="$LIBOIL_CFLAGS -D_GNU_SOURCE -D_POSIX_C_SOURCE=200112L -I\$(top_srcdir) -O2"
+LIBOIL_CFLAGS="$LIBOIL_CFLAGS -D_GNU_SOURCE -D_POSIX_C_SOURCE=200112L -I\$(top_srcdir)"
 AC_SUBST(LIBOIL_CFLAGS)
 
 LIBOIL_LIBS="\$(top_builddir)/liboil/liboil-$LIBOIL_MAJORMINOR.la"
@@ -172,9 +173,6 @@
 pkgconfigdir="\$(libdir)/pkgconfig"
 AC_SUBST(pkgconfigdir)
 
-#CFLAGS=`echo "$CFLAGS" | sed -e 's/-O[0-9*]//g'`
-CFLAGS="-g"
-
 AC_CONFIG_FILES([
 Makefile 
 doc/Makefile
