--- liboil-0.3.6/liboil/copy/splat_ref.c.orig	2006-05-17 01:11:44.708203000 +0200
+++ liboil-0.3.6/liboil/copy/splat_ref.c	2006-05-17 02:06:20.831568000 +0200
@@ -100,19 +100,7 @@
 
 static void splat_u8_ns_int (uint8_t *dest, uint8_t *param, int n)
 {
-  int p;
-  while(n&3) {
-    *dest = *param;
-    dest++;
-    n--;
-  }
-  n >>= 2;
-  p = (*param<<24) | (*param<<16) | (*param<<8) | (*param);
-  while(n>0){
-    *(uint32_t *)dest = p;
-    dest+=4;
-    n--;
-  }
+  memset (dest, *param, n);
 }
 OIL_DEFINE_IMPL(splat_u8_ns_int, splat_u8_ns);
 
